core/src/main/scala/org/apache/spark/sql/execution/oplog/impl/OpLogRdd.scala

```diff
@@ -528,6 +529,16 @@ class OpLogRdd(
   override def compute(split: Partition, context: TaskContext): Iterator[Row] = {
     logDebug(s"starting compute for partition ${split.index} of table $fqtnUpper")
     try {
+      val currentHost = SparkEnv.get.executorId
+      val removePattern = "(executor_).*(_)".r
+      var expectedHost  = bucketHostMap.getOrElse(split.index,"")
+      require(expectedHost.nonEmpty, s"No preferred host found.")
+
+      expectedHost = removePattern.replaceAllIn(expectedHost, "")
```

Comment:
Shouldn't we make the expectedHost patten removal while creating the relation itself so that in each rdd compute is not required and code clooks cleaner as well.
Also give corrective action to the user as well in the exception message.

Reply:
Right, makes sense. will move pattern removal in OpLogFormatRelation. And qualify exception message.